File: dataset/ZeroShot/CLAUDE_claude-sonnet-4-5-20250929/Task69.py

import os
import sys
from pathlib import Path

def delete_file(file_path):
    if not file_path or not file_path.strip():
        return "Error: File path cannot be null or empty"
    
    try:
        # Sanitize the file path to prevent path traversal attacks
        path = Path(file_path).resolve()
        
        # Check if file exists
        if not path.exists():
            return f"Error: File does not exist: {file_path}"
        
        # Check if it's a file (not a directory)\n        if not path.is_file():\n            return f"Error: Path is not a file: {file_path}"\n        \n        # Check if file is writable\n        if not os.access(path, os.W_OK):\n            return f"Error: No permission to delete file: {file_path}"\n        \n        # Delete the file\n        os.remove(path)\n        return f"Success: File deleted successfully: {file_path}"\n        \n    except PermissionError:\n        return f"Error: Permission denied: {file_path}"\n    except Exception as e:\n        return f"Error: {str(e)}"\n\ndef main():\n    # Test cases\n    print("Test Case 1: Empty file path")\n    print(delete_file(""))\n    print()\n    \n    print("Test Case 2: None file path")\n    print(delete_file(None))\n    print()\n    \n    print("Test Case 3: Non-existent file")\n    print(delete_file("nonexistent_file.txt"))\n    print()\n    \n    print("Test Case 4: Create and delete a test file")\n    try:\n        test_file = "test_file_to_delete.txt"\n        with open(test_file, 'w') as f:
            f.write("test content")
        print(delete_file(test_file))
    except Exception as e:
        print(f"Error creating test file: {str(e)}")
    print()
    
    print("Test Case 5: Command line argument")
    if len(sys.argv) > 1:
        print(delete_file(sys.argv[1]))
    else:
        print("No command line argument provided")

if __name__ == "__main__":
    main()

File: dataset/ZeroShot/CLAUDE_claude-sonnet-4-5-20250929/Task69.java

import java.io.File;
import java.nio.file.Files;
import java.nio.file.Paths;
import java.nio.file.Path;

public class Task69 {
    public static String deleteFile(String filePath) {
        if (filePath == null || filePath.trim().isEmpty()) {
            return "Error: File path cannot be null or empty";
        }
        
        try {
            // Sanitize the file path to prevent path traversal attacks
            Path path = Paths.get(filePath).normalize().toAbsolutePath();
            File file = path.toFile();
            
            // Check if file exists
            if (!file.exists()) {
                return "Error: File does not exist: " + filePath;
            }
            
            // Check if it's a file (not a directory)
            if (!file.isFile()) {
                return "Error: Path is not a file: " + filePath;
            }
            
            // Check if file is readable/writable
            if (!file.canWrite()) {
                return "Error: No permission to delete file: " + filePath;
            }
            
            // Delete the file
            if (Files.deleteIfExists(path)) {
                return "Success: File deleted successfully: " + filePath;
            } else {
                return "Error: Failed to delete file: " + filePath;
            }
        } catch (Exception e) {
            return "Error: " + e.getMessage();
        }
    }
    
    public static void main(String[] args) {
        // Test cases
        System.out.println("Test Case 1: Empty file path");
        System.out.println(deleteFile(""));
        System.out.println();
        
        System.out.println("Test Case 2: Null file path");
        System.out.println(deleteFile(null));
        System.out.println();
        
        System.out.println("Test Case 3: Non-existent file");
        System.out.println(deleteFile("nonexistent_file.txt"));
        System.out.println();
        
        System.out.println("Test Case 4: Create and delete a test file");
        try {
            File testFile = new File("test_file_to_delete.txt");
            testFile.createNewFile();
            System.out.println(deleteFile("test_file_to_delete.txt"));
        } catch (Exception e) {
            System.out.println("Error creating test file: " + e.getMessage());
        }
        System.out.println();
        
        System.out.println("Test Case 5: Command line argument");
        if (args.length > 0) {
            System.out.println(deleteFile(args[0]));
        } else {
            System.out.println("No command line argument provided");
        }
    }
}
